.github/workflows/third-party.yml

```
@@ -502,6 +502,52 @@ jobs:
         working-directory: ./python
         run: uv run --frozen pytest ./tests/unit
 
+  test-langchain:
+    name: Test LangChain (main branch) on Python ${{ matrix.python-version }}
+    # If 'schedule' was the trigger, don't run it on contributors' forks
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'schedule' && github.repository == 'pydantic/pydantic') ||
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'third-party-tests'))
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.9', '3.10', '3.11', '3.12']
+
+    steps:
+      - name: Checkout Langchain
+        uses: actions/checkout@v4
+        with:
+          repository: langchain-ai/langchain
+
+      - name: Checkout Pydantic
+        uses: actions/checkout@v4
+        with:
+          path: pydantic-latest
+
+      - name: Install UV
+        uses: astral-sh/setup-uv@v5
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
```

```suggestion
      - name: Install UV
        uses: astral-sh/setup-uv@v5
        with:
            python-version: ${{ matrix.python-version }}
```

The reason I had to use the setup-python action in the previous third-party test is because of the comment I added regarding the uv action. In normal circumstances (i.e. when the project isn't nested under a specific repository folder) you can just let uv setup the Python version.